### PR TITLE
feat: dynamic authorization service interfaces added

### PR DESCRIPTION
**Pull request by htztomic**
Description of changes:
Dynamic authorization service stub added along with request and response interfaces.

Checklist:

* [ ] Have you successfully deployed to an AWS account with your changes?
* [ ] Have you written new tests for your core changes, as applicable?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/addUserToGroup.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+export interface AddUserToGroupRequest {
+  /**
+   * {@link AuthenticatedUser}
+   */
+  authenticatedUser: AuthenticatedUser;
+  /**
+   * User id associated to user to be added to group
+   */
+  userId: string;
```
Comment (bhudnell) [inline workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/addUserToGroup.ts:14]: `AuthenticatedUser` has an id param. Is `userId` different? Or can we remove it?

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/dynamicOperation.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+   */
+  subjectId: string;
+  /**
+   * The field a {@link User} wants access to.
+   */
+  field?: string;
+  /**
+   * Attributes associated to the subject
+   */
+  subjectAttributes?: { [key: string]: unknown };
```
Comment (bhudnell) [inline workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/dynamicOperation.ts:26]:
```suggestion
  subjectAttributes?: Record<string, unknown>;
```

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/identityPermission.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+   *  subject: 'Article',
+   *  fields: ['method']
+   * };
+   * ```
+   */
+  fields?: string[];
+  /**
+   * Used to conditionally restrict a {@link User}'s action
+   */
+  conditions?: { [key: string]: unknown };
```
Comment (bhudnell) [inline workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/identityPermission.ts:62]:
```suggestion
  conditions?: Record<string, unknown>;
```

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/isAuthorizedOnRoute.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  /**
+   * {@link HTTPMethod}
+   */
+  method: HTTPMethod;
+
+  /**
+   * Optional params for variable based operations
+   */
+  params?: { [key: string]: unknown };
```
Comment (bhudnell) [inline workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/isAuthorizedOnRoute.ts:26]:
```suggestion
  params?: Record<string, unknown>;
```

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/isAuthorizedOnSubject.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+export interface IsAuthorizedOnSubjectRequest {
+  /**
+   * {@link AuthenticatedUser}
+   */
+  user: AuthenticatedUser;
+  /**
+   * {@link DynamicOperation}
+   */
+  dynamicOperation: DynamicOperation;
+}
```
Comment (bhudnell) [inline workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/isAuthorizedOnSubject.ts:16]: Does this need a response?

**Comment by sevu1** [comment]
Is it possible to move/update existing models? https://github.com/aws-solutions/solution-spark-on-aws/tree/develop/workbench-core/authorization/src/models

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/isAuthorizedOnSubject.ts (hunk 5 above)
Comment (htztomic) [reply workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/isAuthorizedOnSubject.ts:16]: No it is a thrown error

File: workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/addUserToGroup.ts (hunk 1 above)
Comment (htztomic) [reply workbench-core/authorization/src/dynamicAuthorization/dynamicAuthorizationInputs/addUserToGroup.ts:14]: authenticatedUser is used for auditing purposes, the userId added to group isn't the userId requesting the action to do so

**Review by bhudnell — APPROVED** [review]

**Review by sevu1 — APPROVED** [review]

**Pull request merged**